### PR TITLE
Add a JupyterLab extension for the memory usage status bar item

### PR DESCRIPTION
**Pull request by jtpio**
See https://github.com/jupyterlab/jupyterlab/issues/9363

In JupyterLab 3.0, users will still get the memory usage status bar item on `pip install nbresuse`.

Leveraging the federated (also called "pre-built" / "dynamic") extensions in JupyterLab 3.0: https://jupyterlab.readthedocs.io/en/latest/developer/extension_dev.html#goals-of-the-federated-extension-system

~Displays the indicator twice for now, until JupyterLab removes it from the core extensions~ **Edit**: This is not the case anymore when updating to `0.4.0` without enabling the `/metrics` endpoint.

image

### TODO

- [x] Bootstrap the extension
- [x] Port the code from lab
- [x] Enable strict null checks in tsconfig
- [x] Add dev instructions to the readme
- [x] Adopt the monorepo structure from the start
- [x] Rename the extension to `@jupyter-server/resource-usage`
- [x] Credit the original authors. The status bar was added to core in this squashed commit: https://github.com/jupyterlab/jupyterlab/commit/d1585013e85b8787708c3612746a375e60150414#diff-be21886035b24ea232455cba02fb349aa53c352e0bfaa55c5a53410fc616be5b. Here we could have a single commit with multiple `Co-authored-by: name <foo@bar.com>` entries

### CODE REVIEW

File: packages/labextension/package.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  },
+  "license": "BSD-3-Clause",
+  "author": "nbresuse contributors",
+  "files": [
+    "lib/**/*.{d.ts,eot,gif,html,jpg,js,js.map,json,png,svg,woff2,ttf}",
+    "style/**/*.{css,eot,gif,html,jpg,json,png,svg,woff2,ttf}"
+  ],
+  "main": "lib/index.js",
+  "types": "lib/index.d.ts",
+  "style": "style/index.css",
```
Comment (jasongrout) [inline packages/labextension/package.json:22]: We do not need to have a style key, and we can delete the empty style file.

**Comment by jtpio** [comment]
Rebased to include the renaming from `nbresuse` to `jupyter-resource-usage` from in https://github.com/jupyter-server/jupyter-resource-usage/pull/74

**Comment by ajbozarth** [comment]
I wanted to give this a deeper review, but I have other commitments prior to EOY so for now I can say I tested this locally and it it worked as expected and on an initial pass the code look good. So I'll give this a soft approval

**Comment by jtpio** [comment]
Thanks @ajbozarth.

Just pushed a couple more changes to:

- Add the Jupyter Server extension config file: `jupyter server extension list`
- Move the config to the `jupyter-config` folder

**Pull request merged**